### ISSUE TITLE
Potentially confusing scoping behavior in macros

### ISSUE THREAD

**Issue by joehuchette** [JuliaOpt/JuMP.jl#582]
```jl
using JuMP
m = Model()
@defVar(m, x[1:3])

for i in 1:3
    @addConstraint(m, sum{x[i], i=1} == 1)
    @show i
end
```
Gives
```
i = 1
i = 1
i = 1
```
This doesn't qualify as a bug necessarily, but is potentially confusing from a users perspective.

**Comment by mlubin** [comment]
That's a feature of julia, but I think it would be reasonable for the macros to introduce a ``let`` block around for loops and mark any indexing variables as local, if there's no performance penalty. Would require a whole new chunk of code in the macros.

**Comment by joehuchette** [comment]
Closed by #584
